pkg/kubectl/cmd/set/set_resources_test.go

```diff
@@ -0,0 +1,102 @@
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package set
+
+import (
+	"bytes"
+	"net/http"
+	"strings"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/rest/fake"
+	"k8s.io/kubernetes/pkg/api"
+	cmdtesting "k8s.io/kubernetes/pkg/kubectl/cmd/testing"
+	"k8s.io/kubernetes/pkg/printers"
+)
+
+func TestResourceLocal(t *testing.T) {
+	f, tf, codec, ns := cmdtesting.NewAPIFactory()
+	tf.Client = &fake.RESTClient{
+		APIRegistry:          api.Registry,
+		NegotiatedSerializer: ns,
+		Client: fake.CreateHTTPClient(func(req *http.Request) (*http.Response, error) {
+			t.Fatalf("unexpected request: %s %#v\n%#v", req.Method, req.URL, req)
+			return nil, nil
+		}),
+	}
+	tf.Namespace = "test"
+	tf.ClientConfig = &restclient.Config{ContentConfig: restclient.ContentConfig{GroupVersion: &api.Registry.GroupOrDie(api.GroupName).GroupVersion}}
+
+	buf := bytes.NewBuffer([]byte{})
+	cmd := NewCmdResources(f, buf, buf)
+	cmd.SetOutput(buf)
+	cmd.Flags().Set("output", "name")
+	cmd.Flags().Set("local", "true")
+	cmd.Flags().Set("filename", "../../../../examples/storage/cassandra/cassandra-controller.yaml")
+	cmd.Flags().Set("limits", "cpu=500m,memory=1Gi")
+	cmd.Flags().Set("requests", "cpu=200m,memory=500Gi")
+	mapper, typer := f.Object()
+	tf.Printer = &printers.NamePrinter{Decoders: []runtime.Decoder{codec}, Typer: typer, Mapper: mapper}
+
+	cmd.Run(cmd, []string{})
+
+	if !strings.Contains(buf.String(), "replicationcontrollers/cassandra") {
+		t.Errorf("did not set resource: %s", buf.String())
+	}
+}
+
+func TestSetResourceValidation(t *testing.T) {
+	testCases := []struct {
+		name             string
+		resourcesOptions *ResourcesOptions
+		expectErr        string
+	}{
+		{
+			name: "sucess test case",
+			resourcesOptions: &ResourcesOptions{
+				Limits:   "cpu=500m,memory=1Gi",
+				Requests: "cpu=500m,memory=1Gi",
+			},
+			expectErr: "",
+		},
+		{
+			name:             "test for requests and limits empty",
+			resourcesOptions: &ResourcesOptions{},
+			expectErr:        "you must specify an update to requests or limits (in the form of --requests/--limits)",
+		},
+		{
+			name: "test invalid argument",
+			resourcesOptions: &ResourcesOptions{
+				Requests: "cpu=500m memory=1Gi",
+			},
+			expectErr: "Invalid argument syntax cpu=500m memory=1Gi, expected <resource>=<value>",
+		},
+	}
+	for _, testCase := range testCases {
+		err := testCase.resourcesOptions.Validate()
+		if err != nil {
```

Comment:
@zjj2wry sorry for my long silence, I was OOO for quite a while. Anyway, I've tried this:
```go
		if err != nil && err.Error() != testCase.expectErr {
			t.Errorf("[%s]:expect err:%s got err:%s", testCase.name, testCase.expectErr, err.Error())
		}
```
here and it works like expected. Mind updating?